src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingWorkCoordinator.UnitTestingAsyncDocumentWorkItemQueue.cs

```diff
@@ -46,7 +46,11 @@ protected override bool TryTake_NoLock(DocumentId key, out UnitTestingWorkItem w
                 }
 
                 protected override bool TryTakeAnyWork_NoLock(
-                    ProjectId? preferableProjectId, ProjectDependencyGraph dependencyGraph, IDiagnosticAnalyzerService? service,
+                    ProjectId? preferableProjectId,
+#if false // Not used in unit testing crawling
+                    ProjectDependencyGraph dependencyGraph,
+                    IDiagnosticAnalyzerService? service,
```

Comment:
the dependency graph and diagservice are used by solution-crawler to order what files to analyze, preferring files with diagnostics in them (so they can be removed quickly if fixed).  Such a concept is irrelevant for unittest source-discovery.  So this greatly simplifies things and prevents the need for workspace access to compute this data and pass it along.

Reply:
If a change is made in one project, was the dependency graph being used to ensure that the files in this project are analyzed before other projects that depend on it?

Reply:
@shyamnamboodiripad No, we would do this:

```c#
                        foreach (var dependingProjectId in dependencyGraph.GetProjectsThatDirectlyDependOnThisProject(projectId))
                        {
                            if (workQueue.ContainsKey(dependingProjectId) && analyzerService?.ContainsDiagnostics(Workspace, dependingProjectId) == true)
                            {
                                return dependingProjectId;
                            }
                        }
```

So it was only relevant if the projects actually contain diagnostics.  if not, then the dependency graph had no effect.